### PR TITLE
Dts 25767 commitment reliability

### PR DESCRIPTION
**Pull request by pragauta3**

### CODE REVIEW

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/CommittmentReliabilityServiceImpl.java [hunk 1]
```diff
@@ -78,6 +78,8 @@ public class CommittmentReliabilityServiceImpl extends JiraKPIService<Long, List
 	private static final String COMPLETED_ORIGINAL_ESTIMATE = "completedOriginalEstimate";
 	private static final String INITIALISSUE_ORIGINAL_ESTIMATE = "initialIssueOriginalEstimate";
 	private static final String INITIALCMPLTD_ORIGINAL_ESTIMATE = "initialCompletedIssueOriginalEstimate";
+	private static final String SPECIAL_SYMBOL ="#";
+	private static  final String ALL ="Overall";
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/CommittmentReliabilityServiceImpl.java:82]: remove All field here and use CommonConstant.OVERALL existing field

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/CommittmentReliabilityServiceImpl.java [hunk 2]
```diff
@@ -229,34 +233,63 @@ private void sprintWiseLeafNodeValue(Map<String, Node> mapTmp, List<Node> sprint
 			}
 
 			Map<String, Double> commitmentHowerMap = new HashMap<>();
-			Map<String, Long> commitmentMap = getCommitmentMap(totalPresentJiraIssue, totalPresentCompletedIssue,
-					validationDataList, commitmentHowerMap, fieldMapping, totalPresentInitialIssue,
-					totalPresentCompltdInitialIssue);
+			List<String> uniqueIssues = totalPresentJiraIssue.stream().distinct().map(JiraIssue::getTypeName).distinct().collect(Collectors.toList());
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/CommittmentReliabilityServiceImpl.java:236]: use allJiraIssue instant of totalPresentJiraIssue

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/CommittmentReliabilityServiceImpl.java [hunk 3]
```diff
@@ -229,34 +233,63 @@ private void sprintWiseLeafNodeValue(Map<String, Node> mapTmp, List<Node> sprint
...
+			Map<String, List<JiraIssue>> totalPresentJiraIssueGroup = getGroupByAllIssues(totalPresentJiraIssue);
+			Map<String, List<JiraIssue>> totalPresentCompletedIssueGroup = getGroupByAllIssues(totalPresentCompletedIssue);
+			Map<String, List<JiraIssue>> totalPresentInitialIssueGroup = getGroupByAllIssues(totalPresentInitialIssue);
+			Map<String, List<JiraIssue>> totalPresentCompltdInitialIssueGroup = getGroupByAllIssues(totalPresentCompltdInitialIssue);
+			Map<String, Long> commitmentMap= null;
+			Map<String, List<DataCount>> dataCountMap = new HashMap<>();
+		         for(String issues : uniqueIssues ) {
+					 commitmentMap = getCommitmentMap(totalPresentJiraIssueGroup.getOrDefault(issues, new ArrayList<>()), totalPresentCompletedIssueGroup.getOrDefault(issues, new ArrayList<>()),
+							 commitmentHowerMap, fieldMapping, totalPresentInitialIssueGroup.getOrDefault(issues, new ArrayList<>()),
+							 totalPresentCompltdInitialIssueGroup.getOrDefault(issues, new ArrayList<>()),issues);
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/CommittmentReliabilityServiceImpl.java:246]: naming conversation should be issueType instant of issues

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/CommittmentReliabilityServiceImpl.java [hunk 4]
```diff
@@ -373,15 +406,14 @@ private void populateExcelData(String requestTrackerId, List<KPIExcelData> excel
...
-	 * @param validationData
 	 * @param totalPresentInitialIssue
 	 * @param totalPresentCompltdInitialIssue
 	 * @return
 	 */
 	private Map<String, Long> getCommitmentMap(List<JiraIssue> totalJiraIssue, List<JiraIssue> completed,
-			List<CommitmentReliabilityValidationData> validationData, Map<String, Double> commitmentHowerMap,
-			FieldMapping fieldMapping, List<JiraIssue> totalPresentInitialIssue,
-			List<JiraIssue> totalPresentCompltdInitialIssue) {
+											   Map<String, Double> commitmentHowerMap,
```
Comment (hirbabar) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/jira/scrum/service/CommittmentReliabilityServiceImpl.java:414]: do all file formatting

**Review by BhagyashreeShettigar — DISMISSED** [review]

File: database-scripts/upgrade-scripts/Release-8.0-from-7.0.js [hunk 5]
```diff
@@ -212,3 +212,51 @@ bulkUpdateKpiFieldMapping.push({
 if (bulkUpdateKpiFieldMapping.length > 0) {
     db.kpi_fieldmapping.bulkWrite(bulkUpdateKpiFieldMapping);
 }
+
+
+ //7.4 changes
```
Comment (hirbabar) [inline database-scripts/upgrade-scripts/Release-8.0-from-7.0.js:217]: add comment with ticket id and appropriate msg of changes

**Review by hirbabar — APPROVED** [review]

**Review by Chittauri — APPROVED** [review]

**Pull request merged**